pkg/storage/iterator.go

```diff
@@ -415,6 +423,13 @@ func fetchLazyChunks(ctx context.Context, chunks []*chunkenc.LazyChunk) error {
 		}(fetcher, chunks)
 	}
 
+	// remove data of any invalid chunk
+	for fetcher := range invalidChunks {
+		for _, chk := range chksByFetcher[fetcher] {
+			chk.Chunk.Data = nil
+		}
+	}
+
```

Comment:
@cyriltovena Kindly validate above code change before I add testcases and do cleanup.